content/general/push/activate-subscribe.textile

```diff
@@ -44,31 +38,39 @@ Before you can activate your push device or receive push notifications, you must
 
-Registration tokens must be notified to the Ably library by calling @AblyFirebaseInstanceIdService.onNewRegistrationToken()@. If deriving from "@FirebaseInstanceIdService@":https://firebase.google.com/docs/reference/android/com/google/firebase/iid/FirebaseInstanceIdService directly, for example, your class should end up looking more or less like this:
+To push messages to specific devices, Ably uses the device's registration token. Keep Ably updated with the latest device registration token by overriding "@FirebaseMessagingService#onNewToken@":https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/FirebaseMessagingService#public-void-onnewtoken-string-token and calling "@ActivationContext#onNewRegistrationToken@":. Your service extending "FirebaseMessagingService" should look like the following:
 
 ```[android]
-public class MyRegistrationTokenService extends FirebaseInstanceIdService {
-    @Override
-    public void onTokenRefresh() {
-        // Get updated InstanceID token.
-        String token = FirebaseInstanceId.getInstance().getToken();
-        // Notify Ably
-        AblyFirebaseInstanceIdService.onNewRegistrationToken(this, token);
-    }
+public class AblyPushNotificationService extends FirebaseMessagingService {
+  public static final String PUSH_NOTIFICATION_ACTION = AblyPushNotificationService.class.getName() + ".PUSH_NOTIFICATION_MESSAGE";
+
+  @Override
+  public void onMessageReceived(@NonNull RemoteMessage message) {
+      // TODO Handle the message delivered to the device.
+  }
+
+  @Override
+  public void onNewToken(@NonNull String s) {
+    super.onNewToken(s);
+    //Store token in Ably
+    ActivationContext.getActivationContext(this).onNewRegistrationToken(RegistrationToken.Type.FCM, s);
+  }
 }
 ```
 
-h3. Install Ably for Apple Push Notifications on iOS
+h3. On iOS
 
-If you haven't yet, you should first "set up APNs":https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/APNSOverview.html in your app.
+"Create an APNs @.p12@ certificate and upload it to your Ably app's dashboard":https://knowledge.ably.com/how-do-i-obtain-the-apns-certificates-needed-for-ios-push-notifications (App > Notifications tab > Push Notifications Setup > Setup Push Notifications). An Apple Developer Program membership is required to do this. This is used by Ably to authenticate with APNs using "certificate-based authentication":https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/establishing_a_certificate-based_connection_to_apns?language=objc. Then, "register your app with APNs":https://developer.apple.com/documentation/usernotifications/registering_your_app_with_apns?language=objc.
```

Comment:
It's not "your Ably app's dashboard" -> your Dashboard contains your Ably's apps. See earlier comment for example of how to reword this.

Reply:
It may be fixed by bea659f387b2f1c2f848bc72cdebb830f5cdd872, [lmkwyt](https://www.urbandictionary.com/define.php?term=lmkwyt) 🙂.

Reply:
Yes!